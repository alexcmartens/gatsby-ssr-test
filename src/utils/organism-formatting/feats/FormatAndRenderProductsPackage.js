import React from "react"
import formatPageLinks from "../formatPageLinks"
import formatCards from "../formatCards"
import { ProductsPackage } from "terra-one"

const FormatAndRenderProductsPackage = props => {
  const { data = {}, i, locale, location, productPage, currentSpaceId, products, siteIdentifier } = props

  const eyebrow = data?.eyebrow
  const header = data?.header
  const text = data?.text
  const productEyebrow = data?.productEyebrow

  const linkType = data?.cta?.url?.contentType
    ? data?.cta?.url?.contentType
    : data?.cta?.url?.internal?.type
    ? data?.cta?.url?.internal?.type
    : null

  const cta = formatPageLinks(data?.cta, locale, linkType, null, currentSpaceId, products)
  // if there is an image from PIM, we need to use it instead of the user populated image.
  const image = {
    url: data?.pimImage
      ? data?.pimImage?.file?.url
      : data?.image?.image?.file?.url,
    alt: data?.pimImage ? data?.pimImage?.title : data?.image?.image?.title,
  }
  const cards = formatCards(
    data?.productCards,
    locale,
    location,
    null,
    null,
    null,
    null,
    currentSpaceId,
    products,
    siteIdentifier
  )
  const content = {
    eyebrow,
    header,
    text,
    productEyebrow,
    cta,
    cards,
    image,
  }

  return <ProductsPackage content={content} />
}
export default FormatAndRenderProductsPackage
