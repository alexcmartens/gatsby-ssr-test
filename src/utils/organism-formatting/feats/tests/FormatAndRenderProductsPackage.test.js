import React from "react"
import renderer from "react-test-renderer"
import FormatAndRenderProductsPackage from "../FormatAndRenderProductsPackage"

jest.mock("../../formatTheme")

describe("FormatAndRenderProductsPackage", () => {
  it("renders correctly", () => {
    const data = {
      __typename: "ContentfulProductsPackage",
      id: "9c8696bd-7390-5338-b6a0-6c62fe72cd88",
      text: "Anim aliquip aliqua aliqua id qui aute anim reprehen derit tempor velit est sint. Anim aliquip aliqua aliqua id qui aute anim reprehen.",
      header: "Many Products",
      eyebrow: "Many Products",
      cta: {
        text: "External Link ",
        url: {
          text: "External link",
          url: "https://trimble.com",
          internal: { type: "ContentfulExternalLink" },
        },
        external: true,
      },
      image: {
        image: {
          file: {
            url: "//images.ctfassets.net/citn2sn5tdjr/36pRZr0RXiZV9uB9A2fDsM/afdd0af113de5cc0e56168ce428b829f/product.svg",
            fileName: "product.svg",
          },
          title: "product test image",
          description: "",
        },
      },
      productCards: [
        {
          __typename: "ContentfulBoundedCard",
          eyebrow: "Eyebrow Option",
          header: "Product Name",
          cardImage: {
            id: "068ed0a3-df08-5b02-9e3e-251d0f197f3e",
            focalPoint: { focalPoint: null },
            image: {
              title: "industries-natural-resources-quarry-workers-272x180.jpg",
              file: {
                url: "//images.ctfassets.net/citn2sn5tdjr/3pReviqGaCe5T9cNC86TBF/219b6bf9914f09c66343bbb692f54f8e/industries-natural-resources-quarry-workers-272x180.jpg",
                details: { image: { height: 180, width: 272 } },
              },
            },
            __typename: "ContentfulImageWithFocalPoint",
            title: "Aggregates image",
            altText: null,
          },
          textContent: {
            __typename: "ContentfulRichTextContent",
            colorScheme: null,
            content: {
              raw: '{"nodeType":"document","data":{},"content":[{"nodeType":"paragraph","content":[{"nodeType":"text","value":"Lacus nibh mauris sit sodales pharetra viverra vitae cras aliquet. Turpis sed magna tristique.","marks":[],"data":{}}],"data":{}}]}',
              references: [],
            },
          },
          cardLink: {
            text: "About",
            url: {
              text: "Overview",
              url: "our-company/about/overview",
              internal: { type: "ContentfulPageBase" },
            },
            external: false,
          },
        },
        {
          __typename: "ContentfulBoundedCard",
          eyebrow: "Product Eyebrow",
          header: "Header here",
          cardImage: {
            id: "068ed0a3-df08-5b02-9e3e-251d0f197f3e",
            focalPoint: { focalPoint: null },
            image: {
              title: "industries-natural-resources-quarry-workers-272x180.jpg",
              file: {
                url: "//images.ctfassets.net/citn2sn5tdjr/3pReviqGaCe5T9cNC86TBF/219b6bf9914f09c66343bbb692f54f8e/industries-natural-resources-quarry-workers-272x180.jpg",
                details: { image: { height: 180, width: 272 } },
              },
            },
            __typename: "ContentfulImageWithFocalPoint",
            title: "Aggregates image",
            altText: null,
          },
          textContent: {
            __typename: "ContentfulHtmlBlob",
            blob: {
              blob: "Lacus nibh mauris sit sodales pharetra viverra vitae cras aliquet. Turpis sed magna tristique.",
            },
          },
          cardLink: {
            text: "External Link ",
            url: {
              text: "External link",
              url: "https://trimble.com",
              internal: { type: "ContentfulExternalLink" },
            },
            external: true,
          },
        },
        {
          __typename: "ContentfulBoundedCard",
          eyebrow: "Product Eyebrow",
          header: "Header here",
          cardImage: {
            id: "068ed0a3-df08-5b02-9e3e-251d0f197f3e",
            focalPoint: { focalPoint: null },
            image: {
              title: "industries-natural-resources-quarry-workers-272x180.jpg",
              file: {
                url: "//images.ctfassets.net/citn2sn5tdjr/3pReviqGaCe5T9cNC86TBF/219b6bf9914f09c66343bbb692f54f8e/industries-natural-resources-quarry-workers-272x180.jpg",
                details: { image: { height: 180, width: 272 } },
              },
            },
            __typename: "ContentfulImageWithFocalPoint",
            title: "Aggregates image",
            altText: null,
          },
          textContent: {
            __typename: "ContentfulHtmlBlob",
            blob: {
              blob: "Lacus nibh mauris sit sodales pharetra viverra vitae cras aliquet. Turpis sed magna tristique.",
            },
          },
          cardLink: {
            text: "External Link ",
            url: {
              text: "External link",
              url: "https://trimble.com",
              internal: { type: "ContentfulExternalLink" },
            },
            external: true,
          },
        },
        {
          __typename: "ContentfulBoundedCard",
          eyebrow: "Product Eyebrow",
          header: "Header here",
          cardImage: {
            id: "068ed0a3-df08-5b02-9e3e-251d0f197f3e",
            focalPoint: { focalPoint: null },
            image: {
              title: "industries-natural-resources-quarry-workers-272x180.jpg",
              file: {
                url: "//images.ctfassets.net/citn2sn5tdjr/3pReviqGaCe5T9cNC86TBF/219b6bf9914f09c66343bbb692f54f8e/industries-natural-resources-quarry-workers-272x180.jpg",
                details: { image: { height: 180, width: 272 } },
              },
            },
            __typename: "ContentfulImageWithFocalPoint",
            title: "Aggregates image",
            altText: null,
          },
          textContent: {
            __typename: "ContentfulHtmlBlob",
            blob: {
              blob: "Lacus nibh mauris sit sodales pharetra viverra vitae cras aliquet. Turpis sed magna tristique.",
            },
          },
          cardLink: {
            text: "External Link ",
            url: {
              text: "External link",
              url: "https://trimble.com",
              internal: { type: "ContentfulExternalLink" },
            },
            external: true,
          },
        },
        {
          __typename: "ContentfulBoundedCard",
          eyebrow: "Eyebrow Option",
          header: "Product Name",
          cardImage: {
            id: "068ed0a3-df08-5b02-9e3e-251d0f197f3e",
            focalPoint: { focalPoint: null },
            image: {
              title: "industries-natural-resources-quarry-workers-272x180.jpg",
              file: {
                url: "//images.ctfassets.net/citn2sn5tdjr/3pReviqGaCe5T9cNC86TBF/219b6bf9914f09c66343bbb692f54f8e/industries-natural-resources-quarry-workers-272x180.jpg",
                details: { image: { height: 180, width: 272 } },
              },
            },
            __typename: "ContentfulImageWithFocalPoint",
            title: "Aggregates image",
            altText: null,
          },
          textContent: {
            __typename: "ContentfulRichTextContent",
            colorScheme: null,
            content: {
              raw: '{"nodeType":"document","data":{},"content":[{"nodeType":"paragraph","content":[{"nodeType":"text","value":"Lacus nibh mauris sit sodales pharetra viverra vitae cras aliquet. Turpis sed magna tristique.","marks":[],"data":{}}],"data":{}}]}',
              references: [],
            },
          },
          cardLink: {
            text: "About",
            url: {
              text: "Overview",
              url: "our-company/about/overview",
              internal: { type: "ContentfulPageBase" },
            },
            external: false,
          },
        },
        {
          __typename: "ContentfulBoundedCard",
          eyebrow: "Eyebrow Option",
          header: "Product Name",
          cardImage: {
            id: "068ed0a3-df08-5b02-9e3e-251d0f197f3e",
            focalPoint: { focalPoint: null },
            image: {
              title: "industries-natural-resources-quarry-workers-272x180.jpg",
              file: {
                url: "//images.ctfassets.net/citn2sn5tdjr/3pReviqGaCe5T9cNC86TBF/219b6bf9914f09c66343bbb692f54f8e/industries-natural-resources-quarry-workers-272x180.jpg",
                details: { image: { height: 180, width: 272 } },
              },
            },
            __typename: "ContentfulImageWithFocalPoint",
            title: "Aggregates image",
            altText: null,
          },
          textContent: {
            __typename: "ContentfulRichTextContent",
            colorScheme: null,
            content: {
              raw: '{"nodeType":"document","data":{},"content":[{"nodeType":"paragraph","content":[{"nodeType":"text","value":"Lacus nibh mauris sit sodales pharetra viverra vitae cras aliquet. Turpis sed magna tristique.","marks":[],"data":{}}],"data":{}}]}',
              references: [],
            },
          },
          cardLink: {
            text: "About",
            url: {
              text: "Overview",
              url: "our-company/about/overview",
              internal: { type: "ContentfulPageBase" },
            },
            external: false,
          },
        },
        {
          __typename: "ContentfulBoundedCard",
          eyebrow: "Product Eyebrow",
          header: "Header here",
          cardImage: {
            id: "068ed0a3-df08-5b02-9e3e-251d0f197f3e",
            focalPoint: { focalPoint: null },
            image: {
              title: "industries-natural-resources-quarry-workers-272x180.jpg",
              file: {
                url: "//images.ctfassets.net/citn2sn5tdjr/3pReviqGaCe5T9cNC86TBF/219b6bf9914f09c66343bbb692f54f8e/industries-natural-resources-quarry-workers-272x180.jpg",
                details: { image: { height: 180, width: 272 } },
              },
            },
            __typename: "ContentfulImageWithFocalPoint",
            title: "Aggregates image",
            altText: null,
          },
          textContent: {
            __typename: "ContentfulHtmlBlob",
            blob: {
              blob: "Lacus nibh mauris sit sodales pharetra viverra vitae cras aliquet. Turpis sed magna tristique.",
            },
          },
          cardLink: {
            text: "External Link ",
            url: {
              text: "External link",
              url: "https://trimble.com",
              internal: { type: "ContentfulExternalLink" },
            },
            external: true,
          },
        },
        {
          __typename: "ContentfulBoundedCard",
          eyebrow: "Eyebrow Option",
          header: "Product Name",
          cardImage: {
            id: "068ed0a3-df08-5b02-9e3e-251d0f197f3e",
            focalPoint: { focalPoint: null },
            image: {
              title: "industries-natural-resources-quarry-workers-272x180.jpg",
              file: {
                url: "//images.ctfassets.net/citn2sn5tdjr/3pReviqGaCe5T9cNC86TBF/219b6bf9914f09c66343bbb692f54f8e/industries-natural-resources-quarry-workers-272x180.jpg",
                details: { image: { height: 180, width: 272 } },
              },
            },
            __typename: "ContentfulImageWithFocalPoint",
            title: "Aggregates image",
            altText: null,
          },
          textContent: {
            __typename: "ContentfulRichTextContent",
            colorScheme: null,
            content: {
              raw: '{"nodeType":"document","data":{},"content":[{"nodeType":"paragraph","content":[{"nodeType":"text","value":"Lacus nibh mauris sit sodales pharetra viverra vitae cras aliquet. Turpis sed magna tristique.","marks":[],"data":{}}],"data":{}}]}',
              references: [],
            },
          },
          cardLink: {
            text: "About",
            url: {
              text: "Overview",
              url: "our-company/about/overview",
              internal: { type: "ContentfulPageBase" },
            },
            external: false,
          },
        },
        {
          __typename: "ContentfulBoundedCard",
          eyebrow: "Eyebrow Option",
          header: "Product Name",
          cardImage: {
            id: "068ed0a3-df08-5b02-9e3e-251d0f197f3e",
            focalPoint: { focalPoint: null },
            image: {
              title: "industries-natural-resources-quarry-workers-272x180.jpg",
              file: {
                url: "//images.ctfassets.net/citn2sn5tdjr/3pReviqGaCe5T9cNC86TBF/219b6bf9914f09c66343bbb692f54f8e/industries-natural-resources-quarry-workers-272x180.jpg",
                details: { image: { height: 180, width: 272 } },
              },
            },
            __typename: "ContentfulImageWithFocalPoint",
            title: "Aggregates image",
            altText: null,
          },
          textContent: {
            __typename: "ContentfulRichTextContent",
            colorScheme: null,
            content: {
              raw: '{"nodeType":"document","data":{},"content":[{"nodeType":"paragraph","content":[{"nodeType":"text","value":"Lacus nibh mauris sit sodales pharetra viverra vitae cras aliquet. Turpis sed magna tristique.","marks":[],"data":{}}],"data":{}}]}',
              references: [],
            },
          },
          cardLink: {
            text: "About",
            url: {
              text: "Overview",
              url: "our-company/about/overview",
              internal: { type: "ContentfulPageBase" },
            },
            external: false,
          },
        },
      ],
      productEyebrow: "Works with..... ",
    }

    const tree = renderer
      .create(<FormatAndRenderProductsPackage data={data} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
