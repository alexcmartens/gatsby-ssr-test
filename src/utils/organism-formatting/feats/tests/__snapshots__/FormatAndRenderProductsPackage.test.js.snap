// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormatAndRenderProductsPackage renders correctly 1`] = `
<div
  content={
    Object {
      "cards": Array [
        Object {
          "Link": [MockFunction],
          "bgColor": undefined,
          "cardLink": Object {
            "Link": [MockFunction],
            "iconImage": undefined,
            "override": null,
            "setOverride": null,
            "text": "About",
            "url": "/en/our-company/about/overview",
            "video": null,
          },
          "eyebrow": "Eyebrow Option",
          "header": "Product Name",
          "image": Object {
            "alt": "Product Name",
            "focalPointImage": Object {
              "disableImageCompression": undefined,
              "height": 180,
              "title": "industries-natural-resources-quarry-workers-272x180.jpg",
              "url": "//images.ctfassets.net/citn2sn5tdjr/3pReviqGaCe5T9cNC86TBF/219b6bf9914f09c66343bbb692f54f8e/industries-natural-resources-quarry-workers-272x180.jpg",
              "width": 272,
              "x": undefined,
              "y": undefined,
            },
            "wrapperHeight": undefined,
          },
          "modalContent": null,
          "modalType": null,
          "text": Object {
            "data": <RichTextConnector
              data={
                Object {
                  "__typename": "ContentfulRichTextContent",
                  "colorScheme": null,
                  "content": Object {
                    "raw": "{\\"nodeType\\":\\"document\\",\\"data\\":{},\\"content\\":[{\\"nodeType\\":\\"paragraph\\",\\"content\\":[{\\"nodeType\\":\\"text\\",\\"value\\":\\"Lacus nibh mauris sit sodales pharetra viverra vitae cras aliquet. Turpis sed magna tristique.\\",\\"marks\\":[],\\"data\\":{}}],\\"data\\":{}}]}",
                    "references": Array [],
                  },
                }
              }
            />,
            "typename": "ContentfulRichTextContent",
          },
        },
        Object {
          "Link": [MockFunction],
          "bgColor": undefined,
          "cardLink": Object {
            "Link": [MockFunction],
            "iconImage": undefined,
            "override": null,
            "setOverride": null,
            "text": "External Link ",
            "url": "https://trimble.com",
            "video": null,
          },
          "eyebrow": "Product Eyebrow",
          "header": "Header here",
          "image": Object {
            "alt": "Header here",
            "focalPointImage": Object {
              "disableImageCompression": undefined,
              "height": 180,
              "title": "industries-natural-resources-quarry-workers-272x180.jpg",
              "url": "//images.ctfassets.net/citn2sn5tdjr/3pReviqGaCe5T9cNC86TBF/219b6bf9914f09c66343bbb692f54f8e/industries-natural-resources-quarry-workers-272x180.jpg",
              "width": 272,
              "x": undefined,
              "y": undefined,
            },
            "wrapperHeight": undefined,
          },
          "modalContent": null,
          "modalType": null,
          "text": "Lacus nibh mauris sit sodales pharetra viverra vitae cras aliquet. Turpis sed magna tristique.",
        },
        Object {
          "Link": [MockFunction],
          "bgColor": undefined,
          "cardLink": Object {
            "Link": [MockFunction],
            "iconImage": undefined,
            "override": null,
            "setOverride": null,
            "text": "External Link ",
            "url": "https://trimble.com",
            "video": null,
          },
          "eyebrow": "Product Eyebrow",
          "header": "Header here",
          "image": Object {
            "alt": "Header here",
            "focalPointImage": Object {
              "disableImageCompression": undefined,
              "height": 180,
              "title": "industries-natural-resources-quarry-workers-272x180.jpg",
              "url": "//images.ctfassets.net/citn2sn5tdjr/3pReviqGaCe5T9cNC86TBF/219b6bf9914f09c66343bbb692f54f8e/industries-natural-resources-quarry-workers-272x180.jpg",
              "width": 272,
              "x": undefined,
              "y": undefined,
            },
            "wrapperHeight": undefined,
          },
          "modalContent": null,
          "modalType": null,
          "text": "Lacus nibh mauris sit sodales pharetra viverra vitae cras aliquet. Turpis sed magna tristique.",
        },
        Object {
          "Link": [MockFunction],
          "bgColor": undefined,
          "cardLink": Object {
            "Link": [MockFunction],
            "iconImage": undefined,
            "override": null,
            "setOverride": null,
            "text": "External Link ",
            "url": "https://trimble.com",
            "video": null,
          },
          "eyebrow": "Product Eyebrow",
          "header": "Header here",
          "image": Object {
            "alt": "Header here",
            "focalPointImage": Object {
              "disableImageCompression": undefined,
              "height": 180,
              "title": "industries-natural-resources-quarry-workers-272x180.jpg",
              "url": "//images.ctfassets.net/citn2sn5tdjr/3pReviqGaCe5T9cNC86TBF/219b6bf9914f09c66343bbb692f54f8e/industries-natural-resources-quarry-workers-272x180.jpg",
              "width": 272,
              "x": undefined,
              "y": undefined,
            },
            "wrapperHeight": undefined,
          },
          "modalContent": null,
          "modalType": null,
          "text": "Lacus nibh mauris sit sodales pharetra viverra vitae cras aliquet. Turpis sed magna tristique.",
        },
        Object {
          "Link": [MockFunction],
          "bgColor": undefined,
          "cardLink": Object {
            "Link": [MockFunction],
            "iconImage": undefined,
            "override": null,
            "setOverride": null,
            "text": "About",
            "url": "/en/our-company/about/overview",
            "video": null,
          },
          "eyebrow": "Eyebrow Option",
          "header": "Product Name",
          "image": Object {
            "alt": "Product Name",
            "focalPointImage": Object {
              "disableImageCompression": undefined,
              "height": 180,
              "title": "industries-natural-resources-quarry-workers-272x180.jpg",
              "url": "//images.ctfassets.net/citn2sn5tdjr/3pReviqGaCe5T9cNC86TBF/219b6bf9914f09c66343bbb692f54f8e/industries-natural-resources-quarry-workers-272x180.jpg",
              "width": 272,
              "x": undefined,
              "y": undefined,
            },
            "wrapperHeight": undefined,
          },
          "modalContent": null,
          "modalType": null,
          "text": Object {
            "data": <RichTextConnector
              data={
                Object {
                  "__typename": "ContentfulRichTextContent",
                  "colorScheme": null,
                  "content": Object {
                    "raw": "{\\"nodeType\\":\\"document\\",\\"data\\":{},\\"content\\":[{\\"nodeType\\":\\"paragraph\\",\\"content\\":[{\\"nodeType\\":\\"text\\",\\"value\\":\\"Lacus nibh mauris sit sodales pharetra viverra vitae cras aliquet. Turpis sed magna tristique.\\",\\"marks\\":[],\\"data\\":{}}],\\"data\\":{}}]}",
                    "references": Array [],
                  },
                }
              }
            />,
            "typename": "ContentfulRichTextContent",
          },
        },
        Object {
          "Link": [MockFunction],
          "bgColor": undefined,
          "cardLink": Object {
            "Link": [MockFunction],
            "iconImage": undefined,
            "override": null,
            "setOverride": null,
            "text": "About",
            "url": "/en/our-company/about/overview",
            "video": null,
          },
          "eyebrow": "Eyebrow Option",
          "header": "Product Name",
          "image": Object {
            "alt": "Product Name",
            "focalPointImage": Object {
              "disableImageCompression": undefined,
              "height": 180,
              "title": "industries-natural-resources-quarry-workers-272x180.jpg",
              "url": "//images.ctfassets.net/citn2sn5tdjr/3pReviqGaCe5T9cNC86TBF/219b6bf9914f09c66343bbb692f54f8e/industries-natural-resources-quarry-workers-272x180.jpg",
              "width": 272,
              "x": undefined,
              "y": undefined,
            },
            "wrapperHeight": undefined,
          },
          "modalContent": null,
          "modalType": null,
          "text": Object {
            "data": <RichTextConnector
              data={
                Object {
                  "__typename": "ContentfulRichTextContent",
                  "colorScheme": null,
                  "content": Object {
                    "raw": "{\\"nodeType\\":\\"document\\",\\"data\\":{},\\"content\\":[{\\"nodeType\\":\\"paragraph\\",\\"content\\":[{\\"nodeType\\":\\"text\\",\\"value\\":\\"Lacus nibh mauris sit sodales pharetra viverra vitae cras aliquet. Turpis sed magna tristique.\\",\\"marks\\":[],\\"data\\":{}}],\\"data\\":{}}]}",
                    "references": Array [],
                  },
                }
              }
            />,
            "typename": "ContentfulRichTextContent",
          },
        },
        Object {
          "Link": [MockFunction],
          "bgColor": undefined,
          "cardLink": Object {
            "Link": [MockFunction],
            "iconImage": undefined,
            "override": null,
            "setOverride": null,
            "text": "External Link ",
            "url": "https://trimble.com",
            "video": null,
          },
          "eyebrow": "Product Eyebrow",
          "header": "Header here",
          "image": Object {
            "alt": "Header here",
            "focalPointImage": Object {
              "disableImageCompression": undefined,
              "height": 180,
              "title": "industries-natural-resources-quarry-workers-272x180.jpg",
              "url": "//images.ctfassets.net/citn2sn5tdjr/3pReviqGaCe5T9cNC86TBF/219b6bf9914f09c66343bbb692f54f8e/industries-natural-resources-quarry-workers-272x180.jpg",
              "width": 272,
              "x": undefined,
              "y": undefined,
            },
            "wrapperHeight": undefined,
          },
          "modalContent": null,
          "modalType": null,
          "text": "Lacus nibh mauris sit sodales pharetra viverra vitae cras aliquet. Turpis sed magna tristique.",
        },
        Object {
          "Link": [MockFunction],
          "bgColor": undefined,
          "cardLink": Object {
            "Link": [MockFunction],
            "iconImage": undefined,
            "override": null,
            "setOverride": null,
            "text": "About",
            "url": "/en/our-company/about/overview",
            "video": null,
          },
          "eyebrow": "Eyebrow Option",
          "header": "Product Name",
          "image": Object {
            "alt": "Product Name",
            "focalPointImage": Object {
              "disableImageCompression": undefined,
              "height": 180,
              "title": "industries-natural-resources-quarry-workers-272x180.jpg",
              "url": "//images.ctfassets.net/citn2sn5tdjr/3pReviqGaCe5T9cNC86TBF/219b6bf9914f09c66343bbb692f54f8e/industries-natural-resources-quarry-workers-272x180.jpg",
              "width": 272,
              "x": undefined,
              "y": undefined,
            },
            "wrapperHeight": undefined,
          },
          "modalContent": null,
          "modalType": null,
          "text": Object {
            "data": <RichTextConnector
              data={
                Object {
                  "__typename": "ContentfulRichTextContent",
                  "colorScheme": null,
                  "content": Object {
                    "raw": "{\\"nodeType\\":\\"document\\",\\"data\\":{},\\"content\\":[{\\"nodeType\\":\\"paragraph\\",\\"content\\":[{\\"nodeType\\":\\"text\\",\\"value\\":\\"Lacus nibh mauris sit sodales pharetra viverra vitae cras aliquet. Turpis sed magna tristique.\\",\\"marks\\":[],\\"data\\":{}}],\\"data\\":{}}]}",
                    "references": Array [],
                  },
                }
              }
            />,
            "typename": "ContentfulRichTextContent",
          },
        },
        Object {
          "Link": [MockFunction],
          "bgColor": undefined,
          "cardLink": Object {
            "Link": [MockFunction],
            "iconImage": undefined,
            "override": null,
            "setOverride": null,
            "text": "About",
            "url": "/en/our-company/about/overview",
            "video": null,
          },
          "eyebrow": "Eyebrow Option",
          "header": "Product Name",
          "image": Object {
            "alt": "Product Name",
            "focalPointImage": Object {
              "disableImageCompression": undefined,
              "height": 180,
              "title": "industries-natural-resources-quarry-workers-272x180.jpg",
              "url": "//images.ctfassets.net/citn2sn5tdjr/3pReviqGaCe5T9cNC86TBF/219b6bf9914f09c66343bbb692f54f8e/industries-natural-resources-quarry-workers-272x180.jpg",
              "width": 272,
              "x": undefined,
              "y": undefined,
            },
            "wrapperHeight": undefined,
          },
          "modalContent": null,
          "modalType": null,
          "text": Object {
            "data": <RichTextConnector
              data={
                Object {
                  "__typename": "ContentfulRichTextContent",
                  "colorScheme": null,
                  "content": Object {
                    "raw": "{\\"nodeType\\":\\"document\\",\\"data\\":{},\\"content\\":[{\\"nodeType\\":\\"paragraph\\",\\"content\\":[{\\"nodeType\\":\\"text\\",\\"value\\":\\"Lacus nibh mauris sit sodales pharetra viverra vitae cras aliquet. Turpis sed magna tristique.\\",\\"marks\\":[],\\"data\\":{}}],\\"data\\":{}}]}",
                    "references": Array [],
                  },
                }
              }
            />,
            "typename": "ContentfulRichTextContent",
          },
        },
      ],
      "cta": Object {
        "Link": [MockFunction],
        "iconImage": undefined,
        "override": null,
        "setOverride": null,
        "text": "External Link ",
        "url": "https://trimble.com",
        "video": null,
      },
      "eyebrow": "Many Products",
      "header": "Many Products",
      "image": Object {
        "alt": "product test image",
        "url": "//images.ctfassets.net/citn2sn5tdjr/36pRZr0RXiZV9uB9A2fDsM/afdd0af113de5cc0e56168ce428b829f/product.svg",
      },
      "productEyebrow": "Works with..... ",
      "text": "Anim aliquip aliqua aliqua id qui aute anim reprehen derit tempor velit est sint. Anim aliquip aliqua aliqua id qui aute anim reprehen.",
    }
  }
/>
`;
